test_utils/utils.go

```diff
@@ -16,13 +21,81 @@ import (
 
 	"github.com/containernetworking/cni/pkg/ns"
 	"github.com/containernetworking/cni/pkg/types"
+	etcdclient "github.com/coreos/etcd/client"
 	"github.com/onsi/ginkgo"
 	"github.com/onsi/gomega/gexec"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	calnet "github.com/projectcalico/libcalico-go/lib/net"
 	"github.com/vishvananda/netlink"
 )
 
-func PreCreatePool(pool string) string {
-	return Cmd("dist/calicoctl pool add " + pool)
+var kapi etcdclient.KeysAPI
+
+func init() {
+
+	cfg := etcdclient.Config{Endpoints: []string{"http://127.0.0.1:2379"}}
+	c, _ := etcdclient.New(cfg)
+	kapi = etcdclient.NewKeysAPI(c)
+}
+
+// Delete everything under /calico from etcd
+func WipeEtcd() {
+	_, err := kapi.Delete(context.Background(), "/calico", &etcdclient.DeleteOptions{Dir: true, Recursive: true})
+	if err != nil && !etcdclient.IsKeyNotFound(err) {
+		panic(err)
+
+	}
+
+}
+
+func WipeK8sPods() {
+	config, err := clientcmd.DefaultClientConfig.ClientConfig()
+	if err != nil {
+		panic(err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+
+	if err != nil {
+		panic(err)
+	}
+	pods, err := clientset.Pods("test").List(v1.ListOptions{})
+	if err != nil {
+		panic(err)
+	}
+	for _, pod := range pods.Items {
+		err = clientset.Pods("test").Delete(pod.Name, &v1.DeleteOptions{})
+
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func PreCreatePool(pool string) {
```

Comment:
That one's a little more "feature rich". I'm happy with a little duplication for the moment to keep the code simpler.